### PR TITLE
CI: Automate ubuntu version update #15772

### PR DESCRIPTION
**Pull request by salasberryfin**
This PR is related to the following issue #15772 

It adds a new automation that retrieves the latest `ubuntu:focal` version from Docker Hub and updates the Dockerfile in `deploy/kicbase/Dockerfile`. It is configured to run weekly as a GitHub Action and create a new PR, following the procedure found in existing workflows. The image version can also be updated using `make update-ubuntu-version`.

The details of the image are retrieved from the Docker Hub API.

### CODE REVIEW

**Review by spowelljr — CHANGES_REQUESTED** [review]
Looks good, just a couple small things

**Review by spowelljr — CHANGES_REQUESTED** [review]
Also, could you ask this so it will build the kicbase image

https://github.com/kubernetes/minikube/blob/master/.github/workflows/update-docker-version.yml#L51-L60

File: .github/workflows/update-ubuntu-version.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        id: bumpUbuntu
+        run: |
+          make update-ubuntu-version
+          c=$(git status --porcelain)
+          c="${c//$'\n'/'%0A'}"
+          c="${c//$'\r'/'%0D'}"
+          echo "changes<<EOF" >> $GITHUB_OUTPUT
+          echo "$c" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+      - name: Create PR
```
Comment (spowelljr) [inline .github/workflows/update-ubuntu-version.yml:32]: Will need to add:
```
id: createPR
```

So `${{ steps.createPR.outputs.pull-request-number }}` can get the PR number

**Review by spowelljr — CHANGES_REQUESTED** [review]

File: .github/workflows/update-ubuntu-version.yml (hunk 1 above)
Comment (salasberryfin) [reply .github/workflows/update-ubuntu-version.yml:32]: Sorry about that. Took the code snippet from the `update-docker-version` and didn't realize this was missing. I think it is solved now.

**Comment by salasberryfin** [comment]
I applied the proposed changes and I think all of them are resolved now. :)

**Comment by spowelljr** [comment]
Hi @salasberryfin, this looks great! There's one other change that would be good to see, but it can be in a follow up PR if you'd prefer.

It's getting the from and to versions in the PR title, for example our buildkit PR:
```
Kicbase/ISO: Update buildkit from v0.11.4 to v0.11.6
```

If you look at:
https://github.com/kubernetes/minikube/blob/92f364887c20f87d09509d4762c69ccb1cef9d2c/.github/workflows/update-cri-o-version.yml#L25-L27
That's how we get the versions, and then do:
https://github.com/kubernetes/minikube/blob/92f364887c20f87d09509d4762c69ccb1cef9d2c/.github/workflows/update-cri-o-version.yml#L46

The logic for getting the version is in https://github.com/kubernetes/minikube/blob/master/hack/update/get_version/get_version.go

You'd add `ubuntu` to the map and then the regexp to get the version from the `UBUNTU_FOCAL_IMAGE` variable you created.

So the outcome would be your PR title would look like:
```
Kicbase: Update base image from focal-20230308 to focal-20230412
```

If you'd prefer to do this in a follow up PR just let me know and I can merge this one.

**Comment by salasberryfin** [comment]
Hi @spowelljr. If you're okay with it, I'd prefer doing it in a separate PR since I won't be able to look at it before the weekend. I'll be picking it up then and creating a new PR.

**Review by spowelljr — APPROVED** [review]

**Pull request merged**